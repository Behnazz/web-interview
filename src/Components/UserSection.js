import React from 'react'
import PropTypes from 'prop-types'

const UserSection = props => {
  const { firstName, lastName, avatar } = props
  return (
    <span className="app-user">
      <img
        src={avatar}
        className="app-user-logo"
        alt="User Avatar"
        data-testid="avatar"
      />
      <h3 className="user-name">
        {firstName} {lastName}
      </h3>
    </span>
  )
}
UserSection.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  avatar: PropTypes.string,
}

export default UserSection
