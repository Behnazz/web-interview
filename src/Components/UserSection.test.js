import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import UserSection from './UserSection'

afterEach(cleanup)

it('render firstName and lastName', () => {
  const { getByText } = render(
    <UserSection firstName="Behnaz" lastName="testing" />
  )
  expect(getByText('Behnaz testing')).toBeInTheDocument()
})

it('has an avatar', () => {
  const { getByTestId } = render(
    <UserSection avatar="https://www.w3schools.com/howto/img_avatar2.png" />
  )
  expect(getByTestId('avatar')).toBeInTheDocument()
})

it('match the snapshot', () => {
  const { asFragment } = render(<UserSection />)
  expect(asFragment()).toMatchSnapshot()
})
