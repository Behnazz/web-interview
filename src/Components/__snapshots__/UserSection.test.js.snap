// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot 1`] = `
<DocumentFragment>
  <span
    class="app-user"
  >
    <img
      alt="User Avatar"
      class="app-user-logo"
      data-testid="avatar"
    />
    <h3
      class="user-name"
    >
       
    </h3>
  </span>
</DocumentFragment>
`;
